bundles/org.eclipse.ui.ide/extensions/org/eclipse/ui/actions/WorkspaceModifyOperation.java

```diff
@@ -103,20 +103,17 @@ protected abstract void execute(IProgressMonitor monitor)
 	@Override
 	public synchronized final void run(IProgressMonitor monitor)
 			throws InvocationTargetException, InterruptedException {
-		final InvocationTargetException[] iteHolder = new InvocationTargetException[1];
+		final Exception[] exceptionHolder = new Exception[1];
 		try {
 			IWorkspaceRunnable workspaceRunnable = pm -> {
 				try {
 					execute(pm);
-				} catch (InvocationTargetException e1) {
+				} catch (InvocationTargetException | InterruptedException | CoreException e1) {
 					// Pass it outside the workspace runnable
-					iteHolder[0] = e1;
-				} catch (InterruptedException e2) {
-					// Re-throw as OperationCanceledException, which will be
-					// caught and re-thrown as InterruptedException below.
-					throw new OperationCanceledException(e2.getMessage());
+					exceptionHolder[0] = (e1 instanceof CoreException ce) //
+							? new InvocationTargetException(ce) //
+							: e1;
 				}
-				// CoreException and OperationCanceledException are propagated
```

Comment:
I think this comment is rather useful to understand the outside catch, so it should stay.

Reply:
Done in 12615e9efe647076b42e23ad1130bd965296e783